File: backend/src/main/kotlin/me/john/amiscaray/commissionbackend/dto/lobby/LobbyStatus.kt
package me.john.amiscaray.commissionbackend.dto.lobby

import me.john.amiscaray.commissionbackend.dto.game.GameSettings

data class LobbyStatus(var statusType: LobbyStatusType, val subject: Long,
                       var participants: MutableList<Long>, var settings: GameSettings, var host: Long) {


    constructor() : this(LobbyStatusType.DISCONNECT, -1, mutableListOf(), GameSettings(3, 30, 3), -1){



    }

    enum class LobbyStatusType{

        CONNECT,
        DISCONNECT,
        READY,
        NOT_READY,
        KICK_PLAYER,
        START_GAME,
        REQUEST_START

    }

    override fun toString(): String {
        return "LobbyStatus(statusType=$statusType, subject=$subject, participants=$participants, settings=$settings, host=$host)"
    }

}

File: frontend/src/app/dtos/LobbyStatus.ts
import {GameSettings} from "./GameSettings";

export class LobbyStatus{

  constructor(public statusType: string, public subject: number, public host: number=-1,
              public participants: Array<number>=[], public settings: GameSettings=new GameSettings()) {
  }

}

export class LobbyStatusType{

  static CONNECT = 'CONNECT'
  static DISCONNECT = 'DISCONNECT'
  static READY = 'READY'
  static NOT_READY = 'NOT_READY'
  static KICK_PLAYER = 'KICK_PLAYER'
  static REQUEST_START = 'REQUEST_START'
  static START_GAME = 'START_GAME'

}
